package/src/package.rs

```diff
@@ -0,0 +1,305 @@
+pub struct PackageManifest {
+    /// The set of exports in this package.
+    pub exports: BTreeSet<PackageExport>,
+    /// The libraries(packages) linked against by this package, which must be provided when
+    /// executing the program.
+    pub dependencies: Vec<Dependency>,
+}
+
+/// A procedure exported by a package, along with its digest and
+/// signature(will be added after MASM type attributes are implemented).
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Ord, PartialOrd)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct PackageExport {
+    /// The fully-qualified name of the procedure exported by this package
+    pub name: String,
+    /// The digest of the procedure exported by this package
+    #[serde(
+        serialize_with = "se::serialize_digest",
+        deserialize_with = "de::deserialize_digest"
+    )]
+    #[cfg_attr(test, proptest(value = "Digest::default()"))]
+    pub digest: Digest,
+    // Signature will be added in the future when the type signatures are available in the Assembly
+    // #[serde(default)]
+    // pub signature: Option<Signature>,
+}
+
+impl fmt::Debug for PackageExport {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("PackageExport")
+            .field("name", &format_args!("{}", self.name))
+            .field("digest", &format_args!("{}", DisplayHex::new(&self.digest.as_bytes())))
+            // .field("signature", &self.signature)
+            .finish()
+    }
+}
+
+// PACKAGE
+// ================================================================================================
+
+/// A package containing a [Program]/[Library], and a manifest(exports and dependencies).
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct Package {
+    /// Name of the package
+    pub name: String,
+    /// The MAST artifact ([Program] or [Library]) of the package
+    #[serde(serialize_with = "se::serialize_mast", deserialize_with = "de::deserialize_mast")]
+    pub mast: MastArtifact,
+    /// The package manifest, containing the set of exported procedures and their signatures,
+    /// if known.
+    pub manifest: PackageManifest,
+}
+
+impl Package {
+    const MAGIC: &'static [u8] = b"MASP\0";
+    const FORMAT_VERSION: &'static [u8] = b"1.0\0";
+
+    /// Parses a package from the provided bytes
+    pub fn read_from_bytes<B>(bytes: B) -> Result<Self, Report>
+    where
+        B: AsRef<[u8]>,
+    {
+        use alloc::borrow::Cow;
+
+        let bytes = bytes.as_ref();
+
+        let bytes = bytes
+            .strip_prefix(Self::MAGIC)
+            .ok_or_else(|| Report::msg("invalid package: missing header"))?;
+        let bytes = bytes.strip_prefix(Self::FORMAT_VERSION).ok_or_else(|| {
+            Report::msg(format!(
+                "invalid package: incorrect version, expected '1.0', got '{}'",
```

Comment:
small nit: we could use `FORMAT_VERSION` instead of hardcoding `1.0`

Reply:
Done.